.github/workflows/ui.yml

```diff
@@ -0,0 +1,333 @@
+        required: false
+      FOLIO_REGISTRY_PASSWORD:
+        description: Password for publishing module descriptor
+        required: false
+
+jobs:
+  # this job is responsible for printing install information and publishing yarn.lock if debug is enabled
+  install-and-lint:
+    name: Install and lint
+    uses: ./.github/workflows/ui-install-and-lint.yml
+    with:
+      node-version: ${{ inputs.node-version }}
+      folio-npm-registry: ${{ inputs.folio-npm-registry }}
+      allow-lint-errors: ${{ inputs.allow-lint-errors }}
+
+  jest-tests:
+    name: Run Jest tests
+    uses: ./.github/workflows/ui-tests-jest.yml
+    if: ${{ inputs.jest-enabled }}
+    with:
+      node-version: ${{ inputs.node-version }}
+      folio-npm-registry: ${{ inputs.folio-npm-registry }}
+      jest-test-command: ${{ inputs.jest-test-command }}
+      jest-junit-output-dir: ${{ inputs.jest-junit-output-dir }}
+      jest-coverage-report-dir: ${{ inputs.jest-coverage-report-dir }}
+
+  bigtest-tests:
+    name: Run Bigtest tests
+    if: ${{ inputs.bigtest-enabled }}
+    uses: ./.github/workflows/ui-tests-bigtest.yml
+    with:
+      node-version: ${{ inputs.node-version }}
+      folio-npm-registry: ${{ inputs.folio-npm-registry }}
+      bigtest-test-command: ${{ inputs.bigtest-test-command }}
+      bigtest-junit-output-dir: ${{ inputs.bigtest-junit-output-dir }}
+      bigtest-coverage-report-dir: ${{ inputs.bigtest-coverage-report-dir }}
+
+  translations:
+    name: Compile translations
+    if: ${{ inputs.compile-translations }}
+    uses: ./.github/workflows/ui-translations.yml
+    with:
+      node-version: ${{ inputs.node-version }}
+      folio-npm-registry: ${{ inputs.folio-npm-registry }}
+
+  sonarcloud:
+    name: SonarCloud analysis
+    needs: [jest-tests, bigtest-tests]
+    # run even if tests fail
+    if: ${{ !cancelled() }}
+    uses: ./.github/workflows/ui-sonarcloud.yml
+    with:
+      sonar-sources: ${{ inputs.sonar-sources }}
+      sonar-exclusions: ${{ inputs.sonar-exclusions }}
+      bigtest-coverage-report-dir: ${{ inputs.bigtest-coverage-report-dir }}
+      jest-coverage-report-dir: ${{ inputs.jest-coverage-report-dir }}
+    secrets:
+      SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+
+  version-number:
+    # used for module descriptor and publishing to NPM
+    uses: ./.github/workflows/ui-version-number.yml
+
+  generate-module-descriptor:
+    name: Generate module descriptor
+    needs: [version-number]
+    if: ${{ inputs.generate-module-descriptor }}
+    uses: ./.github/workflows/ui-module-descriptor-generate.yml
+    with:
+      node-version: ${{ inputs.node-version }}
+      folio-npm-registry: ${{ inputs.folio-npm-registry }}
+      package-version: ${{ needs.version-number.outputs.version-number }}
+
+  okapi-dependency-check:
+    name: Okapi dependency check (main branch only)
+    needs: [generate-module-descriptor]
+    if: ${{ inputs.generate-module-descriptor && startsWith(github.ref, 'refs/tags/v') }}
```

Comment:
The comment is "main branch only" but the check is actually "tag starting with `v`". These must align. I think the check is correct; agreed? 

Can/should we add "or branch matching  `/^b[0-9]+\.[0-9]+$/` so that this runs both for "tagging a release regardless of branch" and "preparing a commit on a release branch"?

Reply:
Done; consolidated all this logic into a job which calculates these shared variables